modules/update-mco-process.adoc

```diff
@@ -0,0 +1,83 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="mco-update-process_{context}"]
+= Machine Config Operator node updates
+The Machine Config Operator (MCO) applies a new machine configuration to each control plane and compute node. During this machine configuration update, control plane nodes and compute nodes are organized into their own machine config pools, where the pools of machines are updated in parallel to each other. The `.spec.maxUnavailable` configuration, which has a default value of one node, determines how many nodes in a machine config pool can simultaneously undergo the update process at any given time.
+
+When the machine configuration update process begins, the MCO checks the amount of currently unavailable nodes in a pool. If there are fewer unavailable nodes than the value of `.spec.maxUnavailable`, the MCO will initiate the following sequence of actions on available nodes in the pool:
+
+. Cordon and drain the node
++
+[NOTE]
+====
+When a node is cordoned, workloads cannot be scheduled to it.
+====
+
+. Update the system configuration and operating system (OS) of the node
+
+. Reboot the node
+
+. Uncordon the node
+
+A node undergoing this process is unavailable until it is uncordoned and workloads can be scheduled to it again. The MCO begins updating nodes until the number of unavailable nodes is equal to the value of `.spec.maxUnavailable`.
+
+As a node completes its update and becomes available, the number of unavailable nodes in the machine config pool is once again fewer than `.spec.maxUnavailable`. If there are remaining nodes that need to be updated, the MCO initiates the update process on a node until the `.spec.maxUnavailable` limit is once again reached. This process repeats until each control plane and compute node has been updated.
+
+The following example workflow describes how this process might occur in a machine config pool with 5 nodes, where `.spec.maxUnavailable` is 3 and all nodes are initially available:
+
+. The MCO cordons nodes 1, 2, and 3, and begins to drain them.
+
+. Node 2 finishes draining, reboots, and becomes available again. The MCO cordons node 4 and begins draining it.
+
+. Node 1 finishes draining, reboots, and becomes available again. The MCO cordons node 5 and begins draining it.
+
+. Node 3 finishes draining, reboots, and becomes available again.
+
+. Node 5 finishes draining, reboots, and becomes available again.
+
+. Node 4 finishes draining, reboots, and becomes available again.
```

Comment:
I could be real picky here and ask why node 4 finishes draining after 5, despite that 4 started first.  

Or, why node 2 finishes before 1.